solr/core/src/test/org/apache/solr/security/JWTAuthPluginIntegrationTest.java

```diff
@@ -253,45 +239,56 @@ public void createCollectionUpdateAndQueryDistributed() throws Exception {
     result = get(baseUrl + "/" + COLLECTION + "/query?q=*:*", jwtStaticTestToken);
     assertEquals(Integer.valueOf(200), result.second());
     assertAuthMetricsMinimums(10, 10, 0, 0, 0, 0);
-    
+
     // Delete
     assertEquals(200, get(baseUrl + "/admin/collections?action=DELETE&name=" + COLLECTION, jwtStaticTestToken).second().intValue());
     assertAuthMetricsMinimums(11, 11, 0, 0, 0, 0);
     assertPkiAuthMetricsMinimums(4, 4, 0, 0, 0, 0);
+
+    HttpClientUtil.close(cl);
   }
 
+  /**
+   * Configure solr cluster with a security.json talking to MockOAuth2 server
+   * @param numNodes number of nodes in cluster
+   * @param pemFilePath path to PEM file for SSL cert to trust for OAuth2 server
+   * @param timeoutMs how long to wait until the new security.json is applied to the cluster
+   * @return an instance of the created cluster that the test can talk to
+   */
   @SuppressWarnings("BusyWait")
-  private void configureClusterMockOauth(int numNodes, Path pemFilePath, long timeoutMs) throws Exception {
-    configureCluster(numNodes)// nodes
+  private MiniSolrCloudCluster configureClusterMockOauth(int numNodes, Path pemFilePath, long timeoutMs) throws Exception {
+    MiniSolrCloudCluster myCluster = configureCluster(numNodes)// nodes
         .addConfig("conf1", TEST_PATH().resolve("configsets").resolve("cloud-minimal").resolve("conf"))
         .withDefaultClusterProperty("useLegacyReplicaAssignment", "false")
-        .configure();
+        .build();
     String securityJson = createMockOAuthSecurityJson(pemFilePath);
-    cluster.getZkClient().setData("/security.json", securityJson.getBytes(Charset.defaultCharset()), true);
+    myCluster.getZkClient().setData("/security.json", securityJson.getBytes(Charset.defaultCharset()), true);
     RTimer timer = new RTimer();
     do { // Wait timeoutMs time for the security.json change to take effect
       Thread.sleep(200);
       if (timer.getTime() > timeoutMs) {
+        myCluster.shutdown();
```

Comment:
Need to clean up before throwing exception here, since the MiniSolrCloudCluster is already created